components/brave_ads/core/internal/creatives/creative_ad_util.h

```diff
@@ -30,6 +33,27 @@ size_t TargetedCreativeAdCount(const T& creative_ads) {
       });
 }
 
+template <typename T>
+T DeduplicateCreativeAds(const T& creative_ads) {
+  base::flat_set<std::string> uuids;
+
+  T deduplicated_creative_ads;
+  for (const auto& creative_ad : creative_ads) {
+    const std::string uuid =
+        base::StrCat({creative_ad.campaign_id, creative_ad.creative_set_id,
```

Comment:
What do you this about implementing operator== () for necessary classes?
That way we can use a standard std::sort + std::unique to deduplicate this. Or just std::set (or flat_set) as the container

Reply:
Please see other comment concerning draft. Thanks